src/data-workspace/min-max-limits-mutations/use-delete-limits.test.js

```diff
@@ -0,0 +1,363 @@
+                    </Wrapper>
+                ),
+            })
+
+        // we need to wait for the initial data values to have been fetched
+        // (and therefore cached)
+        await waitForDataValueSet(() => {
+            if (dataValueSet.current.isSuccess) {
+                return true
+            }
+
+            if (dataValueSet.current.isError) {
+                return true
+            }
+
+            return false
+        })
+
+        // extract the cached values and make an assertion
+        const dataValueSetQueryKey = dataValueSets.byIds({
+            dataSetId: 'ds-id',
+            periodId: 2022,
+            orgUnitId: 'ou-id',
+            categoryComboId: 'cc-id',
+            categoryOptionIds: ['co-id1', 'co-id2'],
+        })
+        const initialDataValuesQuery = queryCache.find({
+            queryKey: dataValueSetQueryKey,
+        })
+        const initialDataValuesQueryData = initialDataValuesQuery.state.data
+
+        expect(initialDataValuesQueryData).toEqual({
+            minMaxValues: [
+                {
+                    categoryOptionCombo: 'category-option-combo-id',
+                    dataElement: 'data-element-id',
+                    orgUnit: 'orgUnitId',
+                    minValue: 3,
+                    maxValue: 4,
+                },
+            ],
+        })
+
+        const { result, waitFor } = renderHook(useDeleteLimits, {
+            wrapper: ({ children }) => (
+                <Wrapper
+                    queryClientOptions={{ queryCache }}
+                    dataForCustomProvider={{
+                        'dataEntry/minMaxValues': () =>
+                            Promise.reject(new Error('Custom error')),
+                    }}
+                >
+                    {children}
+                </Wrapper>
+            ),
+        })
+
+        act(() => {
+            result.current.mutate({
+                categoryOptionCombo: 'category-option-combo-id',
+                dataElement: 'data-element-id',
+                orgUnit: 'orgUnitId',
+            })
+        })
+
+        await waitFor(() => result.current.isError)
+
+        const dataValuesQuery = queryCache.find({
+            queryKey: dataValueSetQueryKey,
+        })
+        const dataValuesQueryData = dataValuesQuery.state.data
+
+        // The original value is `undefined`, but `queryClient.setQueryData`
+        // doesn't work with `undefined` as a value, so it has to be set to an
+        // empty object instead
+        expect(dataValuesQueryData).toEqual({
+            minMaxValues: [
+                {
+                    categoryOptionCombo: 'category-option-combo-id',
+                    dataElement: 'data-element-id',
+                    orgUnit: 'orgUnitId',
+                    minValue: 3,
+                    maxValue: 4,
+                },
+            ],
+        })
+    })
+
+    it('should pass the correct data to the data engine', async () => {
```

Comment:
I don't see the issue if app-runtime ever changed the implementation details. We'll have to update the hook regardless and the presence of this test won't hide the actual code in any way, does it? It'd require changes to the test, but it's not something I'm worried about, I'd say it's actually good. The intention was to make sure we're sending the correct query to prevent changes to the query that actually won't work, although quite unlikely that that would happen I guess.

You're right, we haven't tested query structure this way yet and I'm not sure if it's really necessary either. But now it's there already and - at least right now - I don't see any issues with it. So I'd just leave it there